Track Default Values

ameily opened the pull request:
This PR adds a `is_default_value` that checks if the configuration value was set to the default. This will return true if config value was set via `__setdefault__` and not overridden by a configuration file or through the API. This method doesn't actual check the *value* of the field. So, if you explicitly set the value to the default, `is_default_value` will return `False`. Also, mutable types, such as list, dict, or a nested config, will return `True` even if a nested field has been modified.

```python
>>> from cincoconfig import Schema, is_default_value, Field
>>> schema = Schema()
>>> schema.x = Field(default=1)
>>> schema.y.z = Field(default=2)
>>> config1 = schema()
>>> is_default_value(config1, 'x')
True
>>> config1.x = 1
>>> is_default_value(config1, 'x')
False
>>> config2 = schema(x=1)
>>> is_default_value(config2, 'x')
False
>>> is_default_value(config2, 'y')
True
>>> config2.y.z = 100
>>> is_default_value(config2, 'y')
True
>>> is_default_value(config2, 'y.z')
False
```

This PR also adds a `reset_value` that can be used to reset a config value to the default.

```python
>>> from cincoconfig import reset_value
>>> config1.x = 100
>>> reset_value(config1, 'x')
>>> config1.x
1
```

Closes #51

ameily commented:
@sradigan let me know what you think and if this aligns with what you were thinking.

sradigan commented:
Yeah man, this looks like exactly what I was thinking! Also nice to have the reset function!

ameily commented:
@sradigan I changed this slightly, I renamed `is_default_value` to `is_value_defined`. The idea was that the previous method name implied that we were checking the actual value against the default. Now, `is_value_defined` is just checking if the value was set by the user, through either loading a configuration file or the API.
